packages/next-server/lib/fetch/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { Response, Request, RequestInit } from 'node-fetch'
+import HttpAgent, { HttpsAgent, HttpOptions } from 'agentkeepalive'
+
+const AGENT_OPTS: HttpOptions = {
+  maxSockets: 200,
+  maxFreeSockets: 20,
+  timeout: 60000, // active socket keepalive for 60 seconds
+  freeSocketKeepAliveTimeout: 30000, // free socket keepalive for 30 seconds
+}
+
+let defaultHttpGlobalAgent: HttpAgent
+let defaultHttpsGlobalAgent: HttpAgent
+
+function getDefaultHttpsGlobalAgent() {
+  return (defaultHttpsGlobalAgent = defaultHttpsGlobalAgent || new HttpsAgent(AGENT_OPTS))
+}
+
+function getDefaultHttpGlobalAgent() {
+  return (defaultHttpGlobalAgent = defaultHttpGlobalAgent || new HttpAgent(AGENT_OPTS))
+}
+
+export type NodeFetch<O = RequestInit> = (
+  url: string | Request,
+  opts?: O,
+) => Promise<Response>
+
+export function getUrl(url: string | Request) {
+  return url instanceof Request ? url.url : url
+}
+
+export function getAgent(url: string | Request) {
+  return /^https/.test(getUrl(url))
```

Comment:
I know we should force `https` by default but still `/^https?/` is better in my opinion

Reply:
`code` is very common in error objects, `status` is used by `res` and users may try that one, and `statusCode` is also very used to get the status code, specially in implementations where `code` and `statusCode` may mean something different.

Reply:
This isn't there to force `https` by default, is only there to choose between the `https` or `http` agent, changing it to `/^https?/` will allow the `https` agent to be used for `http` requests.